utils_linux.go

```diff
@@ -297,15 +310,14 @@ func (r *runner) run(config *specs.Process) (int, error) {
 			return -1, err
 		}
 	}
-	handler := <-handlerCh
-	status, err := handler.forward(process, tty, detach)
-	if err != nil {
-		r.terminate(process)
-	}
 	if detach {
 		return 0, nil
 	}
-	if err == nil {
+	// For non-detached container, we should forward signals to the container.
+	handler := <-handlerCh
+	if status, err = handler.forward(process, tty); err != nil {
+		r.terminate(process)
+	} else {
```

Comment:
This `else` is kinda hard to follow when reading the code. Maybe something like this?

```go
	if status, err = handler.forward(process, tty); err != nil {
		r.terminate(process)
		return -1, err
	}
	r.destroy()
	return status, nil
}
```